test/cred_helper_test.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+import os
+import json
+import unittest
+import subprocess
+
+class CredHelperTests(unittest.TestCase):
+    """ Tests the cred_helper.py script requirements """
+    
+    def setUp(self):
+        """ Sets up unit tests by defining location of script """
+        dir_name = os.path.dirname(__file__)
+        self._exe = os.path.join(dir_name, "..", "cred_helper.py")
+
+    def test_test_mode(self):
+        """
+        Tests that the format of the test mode output is correct and that
+        the user in the .user-bazelrc has the correct entitlements
+        """
+        result = subprocess.run([self._exe, "test"], stdout=subprocess.PIPE,
```

Comment:
this is a non-idiomatic use of unit-tests. It should test that the code is corectly implemented, not that authentication is working.

python mocking could be used to inject the dependencies of subprocess.run() and response.

However, since this has to be tested against a real server anyway, I think three is something to be said for this approach; test that the actual server data is as expected by the appliation.

The actual complexity of this helper is low, so I don't expect it to change after we have tested and weeded out the problems...